### ISSUE TITLE
Exception using CollectionDataSource

### ISSUE THREAD

**Issue by mjmckp** [dotnet/machinelearning#156]
### System information

- Windows 10 Pro 64 Bit
- .NET framework 4.7
- libraries built from master branch

### Issue

The code below (a self-contained repro) throws a null reference exception on line https://github.com/dotnet/machinelearning/blob/83f9bac40939d3fe3485059710f5bd4456bcf02e/src/Microsoft.ML.Api/DataViewConstructionUtils.cs#L233 since `peek` is null when attempting to get the value of the `Label` column.

```
        public class Data
        {
            [ColumnName("Features")]
            [VectorType(2)]
            public float[] Features;

            [ColumnName("Label")]
            public bool Label;
        }

        public class Prediction
        {
            [ColumnName("PredictedLabel")]
            public bool PredictedLabel;
        }

        static void Train(IEnumerable<Data> data)
        {
            var pipeline = new LearningPipeline();
            pipeline.Add(CollectionDataSource.Create(data));
            pipeline.Add(new FastForestBinaryClassifier());
            var model = pipeline.Train<Data, Prediction>();
        }

        static void Main(string[] args)
        {
            var data = new Data[1];
            data[0] = new Data();
            data[0].Features = new float[] { 0.0f, 1.0f };
            data[0].Label = false;

            Train(data);
        }
```

**Comment by mjmckp** [comment]
That's highly unintuitive given that the 1) label type, 2) predicted label type, and 3) classifier output type are all boolean.

**Comment by TomFinley** [comment]
Hi @Ivanidzo4ka are we confident in this diagnosis? I think the trainer can accept bool values?

**Comment by Ivanidzo4ka** [comment]
Should work now, addressed in #183.
